onnx/defs/tensor/defs.cc

```diff
@@ -89,7 +89,98 @@ ONNX_OPERATOR_SET_SCHEMA(
             {"tensor(float16)", "tensor(float)", "tensor(double)"},
             "Constrain input and output types to float tensors.")
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // Type Inference
           propagateElemTypeFromInputToOutput(ctx, 0, 0);
+          // Shape Inference if target shape is in initializer
+          const TensorProto* targetShapeInitializer = ctx.getInputInitializer(1);
+          if (!targetShapeInitializer)  {
+            return;
+          }
+          // Make targetShape (0 -> same as originalShape, -1 -> inferred).
+          // The targetShape vector represents the specified shape for output.
+          std::vector<int64_t> targetShape;
```

Comment:
you can use `data<int64_t>()` to get the tensor's data pointer and size_from_dim(0) to get the #elements

Reply:
Isn't TensorProto different from Tensor?

Reply:
@varunjain99 oh yeah you are right